project/src/main/webapp/gmail.js

```diff
@@ -0,0 +1,365 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+// Array of API discovery doc URLs for APIs used by the quickstart
+var DISCOVERY_DOCS = ["https://www.googleapis.com/discovery/v1/apis/gmail/v1/rest"];
+
+// Authorization scopes required by the API; multiple scopes can be
+// included, separated by spaces.
+var SCOPES = 'https://www.googleapis.com/auth/gmail.readonly';
+
+var authorizeButton = document.getElementById('authorize_button');
+var signoutButton = document.getElementById('signout_button');
+//let emailObjects = {};
+
+/**
+  * On load, called to load the auth2 library and API client library.
+  */
+function handleClientLoad() {
+    gapi.load('client:auth2', initClient);
+}
+
+/**
+  *  Initializes the API client library and sets up sign-in state
+  *  listeners.
+  */
+function initClient() {
+    gapi.client.init({
+        apiKey: API_KEY,
+        clientId: CLIENT_ID,
+        discoveryDocs: DISCOVERY_DOCS,
+        scope: SCOPES
+    }).then(function() {
+        // Listen for sign-in state changes.
+        gapi.auth2.getAuthInstance().isSignedIn.listen(updateSigninStatus);
+
+        // Handle the initial sign-in state.
+        updateSigninStatus(gapi.auth2.getAuthInstance().isSignedIn.get());
+        authorizeButton.onclick = handleAuthClick;
+        signoutButton.onclick = handleSignoutClick;
+    }, function(error) {
+        appendPre(JSON.stringify(error, null, 2));
+    });
+}
+
+/**
+  *  Called when the signed in status changes, to update the UI
+  *  appropriately. After a sign-in, the API is called.
+  */
+function updateSigninStatus(isSignedIn) {
+    if (isSignedIn) {
+        authorizeButton.style.display = 'none';
+        signoutButton.style.display = 'block';
+    } else {
+        authorizeButton.style.display = 'block';
+        signoutButton.style.display = 'none';
+    }
+}
+
+/**
+  *  Sign in the user upon button click.
+  */
+function handleAuthClick(event) {
+    gapi.auth2.getAuthInstance().signIn();
+}
+
+/**
+  *  Sign out the user upon button click.
+  */
+function handleSignoutClick(event) {
+    gapi.auth2.getAuthInstance().signOut();
+}
+
+/**
+  * Append a pre element to the body containing the given message
+  * as its text node. Used to display the results of the API call.
+  *
+  * @param {string} message Text to be placed in pre element.
+  */
+function appendPre(message) {
+    var pre = document.getElementById('content');
+    var textContent = document.createTextNode(message + '\n');
+    pre.appendChild(textContent);
+}
+
+//Get the gmail search query from the front end
+function getQuery() {
+    const query = document.getElementById("text-input").value;
+    console.log("This is the textarea: " + query);
+    return query;
+}
```

Comment:
Can you put the auth code in either the gmail class or a new class/new file?

Reply:
I think it's possible to put it in a new file. I would have to explore more to see if it's possible to put it in the gmail class. Do you think it's necessary to put it in a gmail class if we're not using global variables for the functions?

Reply:
This is done! Thanks for your help today!